File: Java Script - Introduction to Programming/9.1. Problems for Champions - Part I/02. Magic Dates.js
function magicDates([...arg]) {

    let [firstYear, secondYear, numberToSearchFor] = [...arg.map(Number)]
    let date = new Date(firstYear, 0, 1)
    let found = false
    while (date.getFullYear() <= secondYear) {

        let d1 = Math.floor(date.getDate() / 10); // First day digit
        let d2 = date.getDate() % 10; // Second day digit
        let d3 = Math.floor((date.getMonth() + 1) / 10); // First month digit
        let d4 = (date.getMonth() + 1) % 10; // Second month digit
        let d5 = Math.floor(date.getFullYear() / 1000); // First year digit
        let d6 = Math.floor(date.getFullYear() / 100) % 10; // Second year digit
        let d7 = Math.floor(date.getFullYear() / 10) % 10; // Third year digit
        let d8 = date.getFullYear() % 10; // Fourth year digit

        let weight = d1 * (d2 + d3 + d4 + d5 + d6 + d7 + d8) +
            d2 * (d3 + d4 + d5 + d6 + d7 + d8) +
            d3 * (d4 + d5 + d6 + d7 + d8) +
            d4 * (d5 + d6 + d7 + d8) +
            d5 * (d6 + d7 + d8) +
            d6 * (d7 + d8) +
            d7 * d8;

        if (weight == numberToSearchFor) {
            console.log("" + d1 + d2 + "-" + d3 + d4 + "-" + d5 + d6 + d7 + d8);
            found = true;
        }
        date.setDate(date.getDate() + 1);
    }
    if (!found) {
        console.log("No");
    }
}

// magicDates([2012, 2014, 80])
// magicDates([2003, 2004, 1500])

File: Pyrhon - Introduction to Programming/9.1. Problems for Champions - Part I/02. Magic Dates.py
from datetime import date as datetype
from datetime import timedelta
from math import floor

first_year, second_year, number_to_search_for = [int(input()) for _ in range(3)]

date = datetype(first_year, 1, 1)
found = False

while date.year <= second_year:
    d1 = floor(date.day / 10)
    d2 = date.day % 10

    d3 = floor(date.month / 10)
    d4 = date.month % 10

    d5 = floor(date.year / 1000)
    d6 = floor(date.year / 100) % 10
    d7 = floor(date.year / 10) % 10
    d8 = date.year % 10

    weight = d1 * (d2 + d3 + d4 + d5 + d6 + d7 + d8) + \
             d2 * (d3 + d4 + d5 + d6 + d7 + d8) + \
             d3 * (d4 + d5 + d6 + d7 + d8) + \
             d4 * (d5 + d6 + d7 + d8) + \
             d5 * (d6 + d7 + d8) + \
             d6 * (d7 + d8) + \
             d7 * d8

    if weight == number_to_search_for:
        print(str(d1) + str(d2) + "-" + str(d3) + str(d4) +
              "-" + str(d5) + str(d6) + str(d7) + str(d8))
        found = True
    date = date + timedelta(days=1)

if not found:
    print("No")


'''
2007
2007
144

#17-03-2007
#13-07-2007
#31-07-2007
'''
